doc/introduction/new_opmath.rst

```diff
@@ -0,0 +1,436 @@
+| ``qml.ops.LinearCombination(coeffs, ops)`` | n/a        | ``ops.LinearCombination`` |
++----------------------------------+----------------------+---------------------------+
+
+
+The three main new opmath classes ``SProd``, ``Prod``, and ``Sum`` have already been around for a while.
+E.g. ``qml.dot()`` has always returned a ``Sum`` instance.
+
+Usage
+~~~~~
+
+Besides the python operators, you can also use the constructors :func:`~s_prod`, :func:`~prod`, and :func:`~sum`.
+For composite operators, we can access their constituens via the ``op.operands`` attribute.
+
+>>> op = qml.sum(X(0), X(1), X(2))
+>>> op.operands
+(X(0), X(1), X(2))
+
+In case all terms are composed of operators with a valid ``pauli_rep``, then also the composite
+operator has a valid ``pauli_rep`` in terms of a :class:`~pauli.PauliSentence` instance. This is often handy for fast
+arithmetic processing.
+
+>>> op.pauli_rep
+1.0 * X(0)
++ 1.0 * X(1)
++ 1.0 * X(2)
+
+Further, composite operators can be simplified using :func:`~pennylane.simplify` or the ``op.simplify()`` method.
+
+>>> op = 0.5 * X(0) + 0.5 * Y(0) - 1.5 * X(0) - 0.5 * Y(0) # no simplification by default
+>>> op.simplify()
+-1.0 * X(0)
+>>> qml.simplify(op)
+-1.0 * X(0)
+
+Note that the simplification never happens in-place, such that the original operator is left unaltered.
+
+>>> op
+(
+    0.5 * X(0)
+  + 0.5 * Y(0)
+  + -1 * 1.5 * X(0)
+  + -1 * 0.5 * Y(0)
+)
+
+We are often interested in obtaining a list of coefficients and `pure` operators.
+We can do so by using the ``op.terms()`` method.
+
+>>> op = 0.5 * (X(0) @ X(1) + Y(0) @ Y(1) + 2 * Z(0) @ Z(1)) - 1.5 * I() + 0.5 * I()
+>>> op.terms()
+([0.5, 0.5, 1.0, -1.0], [X(1) @ X(0), Y(1) @ Y(0), Z(1) @ Z(0), I()])
+
+As seen by this example, this method already takes care of arithmetic simplifications.
+
+qml.Hamiltonian
+~~~~~~~~~~~~~~~
+
+The legacy classes :class:`~operation.Tensor` and :class:`~ops.Hamiltonian` will soon be deprecated.
+:class:`~ops.LinearCombination` offers the same API as :class:`~ops.Hamiltonian` but works well with new opmath classes.
+
+Depending on whether or not new opmath is active, ``qml.Hamiltonian`` will return either of the two classes.
+
+>>> import pennylane as qml
+>>> from pennylane import X
+>>> qml.operation.active_new_opmath()
+True
+>>> H = qml.Hamiltonian([0.5, 0.5], [X(0), X(1)])
+>>> type(H)
+pennylane.ops.op_math.linear_combination.LinearCombination
+
+>>> qml.operation.disable_new_opmath_()
+>>> qml.operation.active_new_opmath()
+False
+>>> H = qml.Hamiltonian([0.5, 0.5], [X(0), X(1)])
+>>> type(H)
+pennylane.ops.qubit.hamiltonian.Hamiltonian
+
+Both classes offer the same API and functionality, so a user does not have to worry about those implementation details.
+
+.. _Troubleshooting:
+
+Troubleshooting
+---------------
+
+If you are a developer or power-user that explicitly uses ``qml.operation.Tensor`` or ``qml.ops.Hamiltonian``, you
+may run into one of the following common issues.

```

Comment:
Maybe we could phrase it more generally like we did for QNode returns? https://docs.pennylane.ai/en/stable/introduction/returns.html#troubleshooting